src/panels/lovelace/views/hui-view.ts

```diff
@@ -155,33 +126,29 @@ export class HUIView extends UpdatingElement {
           element.hass = this.hass;
         });
 
-        this._layoutElement!.hass = this.hass;
+        this._layoutElement.hass = this.hass;
       }
       if (changedProperties.has("narrow")) {
-        this._layoutElement!.narrow = this.narrow;
+        this._layoutElement.narrow = this.narrow;
       }
       if (changedProperties.has("lovelace")) {
-        this._layoutElement!.lovelace = this.lovelace;
+        this._layoutElement.lovelace = this.lovelace;
       }
       if (changedProperties.has("_cards")) {
-        this._layoutElement!.cards = this._cards;
+        this._layoutElement.cards = this._cards;
       }
       if (changedProperties.has("_badges")) {
-        this._layoutElement!.badges = this._badges;
+        this._layoutElement.badges = this._badges;
       }
     }
 
     const oldHass = changedProperties.get("hass") as this["hass"] | undefined;
 
-    // Update theme if necessary:
-    // - If config changed, the theme could have changed
-    // - if hass themes preferences have changed
     if (
-      configChanged ||
-      (changedProperties.has("hass") &&
-        (!oldHass ||
-          this.hass.themes !== oldHass.themes ||
-          this.hass.selectedTheme !== oldHass.selectedTheme))
+      changedProperties.has("hass") &&
+      (!oldHass ||
+        this.hass.themes !== oldHass.themes ||
+        this.hass.selectedTheme !== oldHass.selectedTheme)
     ) {
       applyThemesOnElement(
```

Comment:
We now don't set the theme on init, but only when the theme changes, we should also do this in `_initializeConfig`.